### PR TITLE
update deploy with k8s on site to use kubernetes provider

### LINKED ISSUE #316: k8s tutorial on secretless site uses k8s secret provider
Depends on #175

### PR DESCRIPTION
**Pull request by doodlesbykumbi**
Resolves #316

also fixes inconsistencies between site and demo

### CODE REVIEW

File: demos/k8s-demo/02_configure_db.sh [hunk 1]
```diff
@@ -26,51 +27,34 @@ CREATE TABLE pets (
...
-  kind: Role
-  name: quick-start-backend-credentials-reader
-  apiGroup: rbac.authorization.k8s.io
-EOL
+# create application service account
+kubectl --namespace quick-start-application-ns \
+  create serviceaccount \
+  quick-start-application
+
+# grant quick-start-application service account in quick-start-application-ns namespace access to quick-start-backend-credentials
```
Comment (sgnn7) [inline demos/k8s-demo/02_configure_db.sh]: Line too long

File: docs/docs/get_started/deploy_to_kubernetes.md [hunk 2]
```diff
@@ -56,23 +58,23 @@ To run through this tutorial, you will need:
...
 
-Pet data is stored in a PostgreSQL database, and the application may be configured to connect to the database by setting the `DB_URL` environment variable in the application's environment (following [12-factor principles](https://12factor.net/)).
+Pet data is stored in a PostgreSQL database, and the application may be configured to connect to the database by setting the `$DB_URL` environment variable in the application's environment (following [12-factor principles](https://12factor.net/)).
 
 See [Try The Running Application](#try-the-running-application) for examples of consuming the routes using `curl`.
 
-The database is **credential-protected**. With the Secretless Broker, we will be able to use a value of `DB_URL` of the form:  `postgresql://x@localhost:5432/${APPLICATION_DB_NAME}?sslmode=disable`. The application will not require knowledge of credentials to connect to the database.
+The database is **credential-protected**. With the Secretless Broker, we will be able to use a value of `$DB_URL` of the form:  `postgresql://localhost:5432/${APPLICATION_DB_NAME}?sslmode=disable`. The application will not require knowledge of credentials to connect to the database.
```
Comment (sgnn7) [inline docs/docs/get_started/deploy_to_kubernetes.md]: s/"`$DB_URL`"/"`DB_URL` environment variable". Ditto for a few other examples of this in this file.

File: demos/k8s-demo/02_configure_db.sh (hunk 1 above)
Comment (doodlesbykumbi) [reply demos/k8s-demo/02_configure_db.sh]: addressed

File: docs/docs/get_started/deploy_to_kubernetes.md (hunk 2 above)
Comment (doodlesbykumbi) [reply docs/docs/get_started/deploy_to_kubernetes.md]: addressed

**Pull request merged**